Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp

```diff
@@ -84,12 +84,7 @@ static String escape_regexp_pattern(const RegExpObject& regexp_object)
     if (pattern.is_empty())
         return "(?:)";
     // FIXME: Check u flag and escape accordingly
-    pattern.replace("\n", "\\n", true);
-    pattern.replace("\r", "\\r", true);
-    pattern.replace(LINE_SEPARATOR_STRING, "\\u2028", true);
-    pattern.replace(PARAGRAPH_SEPARATOR_STRING, "\\u2029", true);
-    pattern.replace("/", "\\/", true);
-    return pattern;
+    return pattern.replace("\n", "\\n", true).replace("\r", "\\r", true).replace(LINE_SEPARATOR_STRING, "\\u2028", true).replace(PARAGRAPH_SEPARATOR_STRING, "\\u2029", true).replace("/", "\\/", true);
```

Comment:
Would an API that takes a vector of things to replace make more sense than chaining like this?
This seems acceptable as is, just thought I'd bring it up.

Reply:
Hmm, the issue is that it would be 2 vectors, a vector of substrings to search and a vector of replacements, where each idx maps from the first to the second, which isnt exactly a nice api

Reply:
I was thinking more Vector<Tuple<StringView,StringView>> or something of to that effect, but we don't need to design an API in this PR.

Reply:
We don't actually have a Tuple class right now, so definitely another PR, but that is indeed a nice API.

Reply:
https://github.com/SerenityOS/serenity/blob/master/AK/Tuple.h ?

Reply:
How did I manage to miss that? :thinking: